tests/test_components/test_base.py

```diff
@@ -113,3 +140,121 @@ def test_load_aea_package_twice():
+
+
+class TestLoadingErrorWithUnusedDependency(AEATestCase):
+    """Test that an error is raised when not used dependencies are detected in import statements of modules."""
+
+    path_to_aea = Path(CUR_PATH) / "data" / "dummy_aea"
+
+    matching_content = re.escape(
+        dedent(
+            """        aea.exceptions.AEAPackageLoadingError: The following dependencies are not used in any Python module of the package (skill, dummy_author/dummy:0.1.0):
+        - connection fetchai/local
+        - protocol fetchai/fipa
+"""
+        )
+    )
+
+    def _add_unused_dependency_in_dummy_skill_imports(self):
+        """
+        Set up the dummy skill for the test.
+
+        Add a dummy dependency that is not used in the skill.
+        """
+        cwd = self._get_cwd()
+        configuration_path = Path(cwd, "skills", "dummy", "skill.yaml")
+        configuration_content = configuration_path.read_text()
+        configuration_content = configuration_content.replace(
+            "protocols:", "protocols:\n- fetchai/fipa:1.0.0"
+        )
+        configuration_content = configuration_content.replace(
+            "connections: []", "connections:\n- fetchai/local:0.20.0"
+        )
+        configuration_path.write_text(configuration_content)
+        self.run_cli_command("fingerprint", "by-path", "skills/dummy", cwd=cwd)
+
+    def test_run(self):
+        """Run the test."""
+        self._add_unused_dependency_in_dummy_skill_imports()
+        with cd(self._get_cwd()):
+            builder = AEABuilder.from_aea_project(Path(self._get_cwd()))
+            with pytest.raises(AEAPackageLoadingError, match=self.matching_content):
+                builder.build()
+
+
+@pytest.mark.parametrize(
+    "component_type,config_file_path",
+    itertools.chain.from_iterable(
+        [
+            zip_longest([], files, fillvalue=component_type)
+            for files, component_type in [
+                (protocol_config_files, PackageType.PROTOCOL),
+                (contract_config_files, PackageType.CONTRACT),
+                (connection_config_files, PackageType.CONNECTION),
+                (skill_config_files, PackageType.SKILL),
+            ]
+        ]
+    ),
+)
+def test_load_all_aea_protocol_packages(
+    component_type: PackageType, config_file_path: str
+) -> None:
+    """Load all AEA component packages."""
+
+    to_be_skipped = {os.path.join(CUR_PATH, "data", "gym-connection.yaml")}
+
+    if config_file_path in to_be_skipped:
+        pytest.skip(
+            f"test not supported for configurations outside package: {config_file_path}"
+        )
+
+    configuration_loader = ConfigLoader.from_configuration_type(
+        component_type, PACKAGE_TYPE_TO_CONFIG_CLASS
+    )
+    with open_file(config_file_path) as fp:
+        configuration_object = configuration_loader.load(fp)
+        directory = Path(config_file_path).parent
+        configuration_object.directory = directory
+        load_aea_package(configuration_object)
```

Comment:
@DavidMinarsch It might be worth considering copying this test in other repositories with a local registry (`packages/` folder), e.g. `open-autonomy`